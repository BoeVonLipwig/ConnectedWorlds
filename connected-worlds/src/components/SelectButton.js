import React from 'react';

class SelectButton extends React.Component {
  render() {
    return (
      <div className="control">
        <div className="checkbox-round">
          <input id={this.props.id} type="checkbox" />
          <label htmlFor={this.props.id} />
        </div>

        <h2>{this.props.name}</h2>
      </div>
    );
  }
}

export default SelectButton;
