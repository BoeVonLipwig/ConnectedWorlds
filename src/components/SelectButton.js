import React from "react";

class SelectButton extends React.Component {
  render() {
    return (
      <div className="control">
        <div className="checkbox-round">
          <input
            id={this.props.id}
            type="checkbox"
            checked={this.props.isChecked}
          />
          <label htmlFor={this.props.id} onClick={this.props.clickHandler} />
        </div>
        <h2>{this.props.name}</h2>
      </div>
    );
  }
}

export default SelectButton;
