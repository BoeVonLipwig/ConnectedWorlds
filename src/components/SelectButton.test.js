import React from "react";
import ReactDOM from "react-dom";
import SelectButton from "./SelectButton";
import { shallow } from "enzyme";

it("renders button without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly as correct name 'Projects'", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton name="Projects" />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly as correct name 'Programme'", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton name="Programme" />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly as correct name 'Collaborators'", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton name="Collaborators" />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly as correct name 'Show Details'", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton name="Show Details" />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly as correct name 'Show Projects' with id 'showProjects'", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SelectButton id="showProjects" name="Projects" />, div);
  expect(div).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(div);
});

it("should have correct checked value of true", () => {
  const wrapper = shallow(<SelectButton id="showSchools" isChecked={true} />);
  expect(wrapper.find("input").props().checked).toBe(true);
});

it("should have correct checked value of false", () => {
  const wrapper = shallow(<SelectButton id="showSchools" isChecked={false} />);
  expect(wrapper.find("input").props().checked).toBe(false);
});

it("should have correct checked value of true to not equal false", () => {
  const wrapper = shallow(<SelectButton id="showSchools" isChecked={true} />);
  expect(wrapper.find("input").props().checked).not.toBe(false);
});

it("should have correct click handler", () => {
  const mockCallback = jest.fn();
  const wrapper = shallow(
    <SelectButton
      id="showSchools"
      isChecked={true}
      clickHandler={mockCallback}
    />
  );
  expect(mockCallback.mock.calls.length).toBe(0);
  wrapper.find("label").simulate("click");
  expect(mockCallback.mock.calls.length).toBe(1);
});
