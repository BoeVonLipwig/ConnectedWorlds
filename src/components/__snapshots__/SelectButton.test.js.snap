// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly as correct name 'Collaborators' 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        type="checkbox"
        readonly=""
      />
      <label />
    </div>
    <h2>
      Collaborators
    </h2>
  </div>
</div>
`;

exports[`renders button correctly as correct name 'Programme' 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        type="checkbox"
        readonly=""
      />
      <label />
    </div>
    <h2>
      Programme
    </h2>
  </div>
</div>
`;

exports[`renders button correctly as correct name 'Projects' 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        type="checkbox"
        readonly=""
      />
      <label />
    </div>
    <h2>
      Projects
    </h2>
  </div>
</div>
`;

exports[`renders button correctly as correct name 'Show Details' 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        type="checkbox"
        readonly=""
      />
      <label />
    </div>
    <h2>
      Show Details
    </h2>
  </div>
</div>
`;

exports[`renders button correctly as correct name 'Show Projects' with id 'showProjects' 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        id="showProjects"
        type="checkbox"
        readonly=""
      />
      <label
        for="showProjects"
      />
    </div>
    <h2>
      Projects
    </h2>
  </div>
</div>
`;

exports[`renders button correctly without crashing 1`] = `
<div>
  <div
    class="control"
  >
    <div
      class="checkbox-round"
    >
      <input
        type="checkbox"
        readonly=""
      />
      <label />
    </div>
    <h2 />
  </div>
</div>
`;
